library/src/commonMain/kotlin/de/westnordost/osm_legal_default_speeds/LegalDefaultSpeeds.kt

```diff
@@ -255,10 +257,11 @@ private fun MutableMap<String,String>.limitSpeedsTo(key: String, maxspeed: Doubl
         }
     }
     /* recurse down. The same should be done for e.g. maxspeed:hgv:conditional if maxspeed:hgv
-    *  already has a lower speed limit etc. */
-    val r = Regex("$key:[a-z_]+")
-    val subkeys = keys.filter { r.matches(it) }
-    for (subkey in subkeys) {
+    *  already has a lower speed limit etc. but copy before as keys will be modified */
+    var copiedKeys = keys.toMutableSet()
+    for (subkey in copiedKeys) {
+        val index = subkey.indexOf("$key:")
+        if (index < 0) continue
```

Comment:
Why not just:
```suggestion
    *  already has a lower speed limit etc. */
    val subkeys = keys.filter { it.startsWith("$key:") }
    for (subkey in subkeys) {
```

Or, (see remark in general review comment)

```suggestion
    *  already has a lower speed limit etc. */
    val subkeys = keys.asSequence().filter { it.startsWith("$key:") }
    for (subkey in subkeys) {
```